src/engine/drpc_client.c

```diff
@@ -214,6 +214,9 @@ ds_get_pool_svc_ranks(uuid_t pool_uuid, d_rank_list_t **svc_ranks)
 	if (ranks == NULL)
 		D_GOTO(out_dresp, rc = -DER_NOMEM);
 
+	D_FREE(gps_resp->svcreps);
+	D_FREE(gps_resp);
```

Comment:
I think more correct will be instead of those D_FREE use the following:
```
out_resp:
              srv__get_pool_svc_resp__free_unpacked(gps_resp, &alloc.alloc);
out_dresp:
```
and change D_GOTO(out_dresp, ...) to D_GOTO(out_dresp, ...) from lines 203, 209 and 215.

Reply:
Yes, that sounds like it should be better.

Reply:
I've added this and re-pushed, let's see if it helps.

L203 does not need this change, but if it did then L200 would also need it I suspect.

Reply:
This worked, thank you.